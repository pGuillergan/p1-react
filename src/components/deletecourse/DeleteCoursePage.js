import React, { useState } from "react";
import { toast } from "react-toastify";
import DeleteCourseForm from "./DeleteCourseForm";
import CoursesPage from "../courses/CoursesPage";

export default function DeleteCoursePage(props) {
  const [course, setCourse] = useState({
    course_title: "",
  });

  function handleChange(event) {
    setCourse({ ...course, [event.target.name]: event.target.value });
  }

  function handleSubmit(event) {
    event.preventDefault();
    fetch("http://54.215.227.33:3000/deletecourse", {
      method: "DELETE",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(course),
    })
      .then((result) => {
        toast.success("Deleted course!");
        props.history.push("/courses");
      })
      .catch((err) => {
        toast.error("Course was not deleted..");
      });
  }

  return (
    <div>
      <div className='jumbotron'>
      <h3>Delete course</h3>
      <DeleteCourseForm
        onSubmit={handleSubmit}
        onChange={handleChange}
        course={course}
      />
      </div>
      <hr/>
      <CoursesPage />
    </div>
  );
}
