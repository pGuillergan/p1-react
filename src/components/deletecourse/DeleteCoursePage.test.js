import React, { Component } from "react";
import { shallow } from "enzyme";

import DeleteCoursePage from "./DeleteCoursePage";

describe("DeletePage", () => {
  it("should delete course", () => {
    const deletePage = shallow(<DeleteCoursePage/>);
    expect(deletePage).toMatchSnapshot();

    expect(deletePage.find('h3').length).toEqual(1);
    expect(deletePage.find('DeleteCourseForm').length).toEqual(1);

  });

});
