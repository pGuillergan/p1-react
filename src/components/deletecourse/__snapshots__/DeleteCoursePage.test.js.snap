// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeletePage should delete course 1`] = `
<div>
  <div
    className="jumbotron"
  >
    <h3>
      Delete course
    </h3>
    <DeleteCourseForm
      course={
        Object {
          "course_title": "",
        }
      }
      onChange={[Function]}
      onSubmit={[Function]}
    />
  </div>
  <hr />
  <CoursesPage />
</div>
`;
